### PR TITLE
Adds e2e tests for Node Autoprovisioning:

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**
This PR adds e2e tests for Node Autoprovisioning:  …
 - should create new node if there is no node for node selector

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -851,6 +847,20 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		By("Check if NAP group was created")
 		Expect(getNAPNodePoolsNumber()).Should(Equal(1))
 	})
+
+	It("should create new node if there is no node for node selector [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		framework.ExpectNoError(enableAutoprovisioning(""))
+		// Create pod allocating 0.7 allocable for present nodes with node selector.
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: s/allocable/allocatable

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -851,6 +847,20 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+		ReserveMemoryWithSelector(f, "memory-reservation", 1, int(0.7*float64(memAllocatableMb)), true, scaleUpTimeout, map[string]string{"test": "test"})
+		defer framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, "memory-reservation")
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: ReserveMemoryWithSelector returns cleanup function, why not use it in defer?

Comment (krzysztof-jastrzebski) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: done

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 3]
```diff
@@ -785,26 +781,26 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+		cleanupFunc1 = nil
 		By("Waiting for scale down to 1")
 		// Verify that cluster size decreased.
 		framework.ExpectNoError(WaitForClusterSizeFunc(f.ClientSet,
 			func(size int) bool { return size == nodeCount+1 }, scaleDownTimeout))
 		By("Delete second pod")
-		pod2Deleted = true
+		cleanupFunc2()
+		cleanupFunc2 = nil
 		framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, "memory-reservation2")
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: delete this if using cleanupFunc2()

Comment (krzysztof-jastrzebski) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: done

**Pull request merged**